sycl/include/syclcompat/device.hpp

```diff
@@ -662,14 +662,64 @@ class dev_mgr {
   unsigned int device_count() { return _devs.size(); }
 
   unsigned int get_device_id(const sycl::device &dev) {
+    if (!_devs.size()) {
+      throw std::runtime_error(
+          "[SYCLcompat] No SYCL devices found in the device list.");
```

Comment:
```suggestion
          "[SYCLcompat] No SYCL devices found in the device list. Device list may have been filtered by syclcompat::filter_device");
```

Reply:
Or something like that. Effectively this is the same error as the `runtime_error` below, just with fewer (none) other devices.